src/dune_named_pipe_stubs/dune

```diff
@@ -0,0 +1,7 @@
+(library
+ (name dune_named_pipe_stubs)
```

Comment:
Why not just `named_pipe`? It's not common to prefix library names with `_stubs`, and it doesn't feel very specific to dune.

Reply:
We should probably introduce a package (`named-pipes`) for this inside dune as well. This needs to be a public component for clients to be written anyway, and the api surface is small enough that it should remain stable.

Reply:
sgtm